docs/design-document.md

```diff
@@ -798,40 +758,29 @@ examples of such invalid entries are:
+clients, these entries cannot be detected prior to them being pushed to the
+synchronization point.
 
 As correctly implemented gittuf clients verify the validity of RSL entries when
-they pull from the main repository, the user is warned if invalid entries are
-encountered. Then, the user can then use M1 to invalidate the incorrect entry.
-Other clients with the invalid entry only need to fetch the latest RSL entries
-to recover. Additionally, the client that created the invalid entries must
-switch to a correct implementation of gittuf before further interactions with
-the main repository.
-
-If the main repository is also gittuf enabled, such incidents can be caught
-before other users receive the incorrect RSL entries. The repository, though,
-must not behave like a typical gittuf client. Instead, gittuf's repository
-behavior is slightly different as RSL entries are submitted before the changes
-they represent. The repository must wait to receive the full changes rather than
-immediately rejecting the RSL entry. TODO: the repository-specific behavior
-needs further discussion.
-
-Consider this example as a representative of this scenario. Bob has a buggy
-gittuf client and pushes an invalid entry to the main repository. Alice pulls
-and receives a warning from her gittuf client. Alice reverses Bob's changes,
-creating an RSL entry for the affected branch if needed, and includes an RSL
-annotation skipping Bob's RSL entry.
-
-#### A gittuf access control policy is violated
-
-Bob creates an RSL entry for a branch he's not authorized for by gittuf policy.
-He pushes a change to that branch. Alice notices this (TODO: decide if alice
-needs to be authorized). Alice reverses Bob's change, creating a new RSL entry
-reflecting that. Alice also creates an RSL annotation marking Bob's entry as
-one to be skipped. Alice, therefore, uses M1.
-
-#### Attacker modifies or deletes historical RSL entry
+they pull from the synchronization point, the user is warned if invalid entries
+are encountered. Then, the user can then use the recovery workflow to invalidate
+the incorrect entry. Other clients with the invalid entry only need to fetch the
+latest RSL entries to recover. Additionally, the client that created the invalid
+entries must switch to a correct implementation of gittuf before further
+interactions with the main repository, but this is left to out-of-band
+synchronization between the actors who notice the issue and the actor using a
+buggy client.
+
+##### A gittuf Access Control Policy is Violated
+
+An actor, Bob, creates an RSL entry for a branch he's not authorized for by
+gittuf policy. He pushes a change to that branch. Another actor, Alice, notices
+this when her gittuf client indicates a failure in the verification workflow.
+Alice creates an RSL annotation marking Bob's entry as one to be skipped. Alice
+also reverses Bob's change, creating a new RSL entry reflecting that. Thus, this
+demonstrates gittuf's recovery workflow in action.
```

Comment:
what does this last sentence really mean?

Reply:
it's unneeded given changes to the recovery flows, taken out.